### PR TITLE
Update arrays.rst

### PR DESCRIPTION
**Pull request by StephenVavasis**
Remark about sub performance

### CODE REVIEW

File: doc/manual/arrays.rst [hunk 1]
```diff
@@ -475,6 +475,14 @@ the same as the result of ``getindex``, except the data is left in place.
...
 can later be used to index the original array indirectly.
 
+The ``sub`` function creates a small data item on the heap to keep track of the
+subarray.  When the subarray itself is small, the
+cost of the heap allocation may overwhelm the benefit of avoidance of the copying.
+In a code involving small subarrays, higher performance may be
+achieved by explicitly
+copying subarrays into and out of temporary storage with for-loops rather than
+invoking ``sub``.
+
```
Comment (omus) [inline doc/manual/arrays.rst]: Should redo the line wrapping

**Comment by mbauman** [comment]
Sorry this fell through the cracks!  I think it could still be worth stating.  Do we have any notes like it anywhere else?

**Comment by StephenVavasis** [comment]
Just wondering... instead of a `for` loop, can we now recommend a dot operation for very small subvectors?   For example, does the following code entail any heap allocation?

```Julia
   u = zeros(2)
    for k = 1 : n  # main loop here
        u[:] .= v[k : k + 1]  # dot op instead of a for-loop
        my_operation!(u)
        v[k : k + 1] .= u  # instead of a for-loop
    end
```

**Comment by KristofferC** [comment]
I think it will allocate a `view`.

**Comment by laborg** [comment]
The statement in this PR isn't true anymore after 1.5 because views no longer forces allocation.

**Pull request closed without merging**